sunpy/coordinates/transformations.py

```diff
@@ -230,6 +231,17 @@ def _check_observer_defined(frame):
                                "but this is valid for only HeliographicCarrington frames.")
 
 
+def _times_are_equal(time_1, time_2):
+    # Checks whether times are equal
+    if isinstance(time_1, Time) and isinstance(time_2, Time):
+        # We explicitly perform the check in TAI to avoid possible numerical precision differences
+        # between a time in UTC and the same time after a UTC->TAI->UTC conversion
+        return np.all(time_1.tai == time_2.tai)
```

Comment:
Out of curiosity: Why do this in TAI rather than with jd1 and jd2?

Reply:
`jd1`/`jd2` is just the format, not the scale.  Comparing `jd1`/`jd2` explicitly – without taking the extra step of forcing the scale conversion to TAI – isn't any better than the previous equality check and would exhibit the same undesirable behavior.

Reply:
Ah right, makes sense.